modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PhpSlim4ServerCodegen.java

```diff
@@ -0,0 +1,368 @@
+                        // .replace("\"", "\\\""));
+
+        // from AbstractPhpCodegen.java
+        // Trim the string to avoid leading and trailing spaces.
+        input = input.trim();
+        try {
+
+            input = URLEncoder.encode(input, "UTF-8")
+                    .replaceAll("\\+", "%20")
+                    .replaceAll("\\%2F", "/")
+                    .replaceAll("\\%7B", "{") // keep { part of complex placeholders
+                    .replaceAll("\\%7D", "}") // } part
+                    .replaceAll("\\%5B", "[") // [ part
+                    .replaceAll("\\%5D", "]") // ] part
+                    .replaceAll("\\%3A", ":") // : part
+                    .replaceAll("\\%2B", "+") // + part
+                    .replaceAll("\\%5C\\%5Cd", "\\\\d"); // \d part
+        } catch (UnsupportedEncodingException e) {
+            // continue
+            LOGGER.error(e.getMessage(), e);
+        }
+        return input;
+    }
+
+    @Override
+    public CodegenOperation fromOperation(String path,
+                                          String httpMethod,
+                                          Operation operation,
+                                          List<Server> servers) {
+        CodegenOperation op = super.fromOperation(path, httpMethod, operation, servers);
+        op.path = encodePath(path);
+        return op;
+    }
+
+    /**
+     * Set PSR-7 implementation package.
+     * Ref: https://www.slimframework.com/docs/v4/concepts/value-objects.html
+     *
+     * @param psr7Implementation PSR-7 implementation package
+     */
+    public void setPsr7Implementation(String psr7Implementation) {
+        switch (psr7Implementation) {
+            case "slim-psr7":
+            case "nyholm-psr7":
+            case "guzzle-psr7":
+            case "zend-diactoros":
+                this.psr7Implementation = psr7Implementation;
+                break;
+            default:
+                this.psr7Implementation = "slim-psr7";
+                LOGGER.warn("\"" + (String) psr7Implementation + "\" is invalid \"psr7Implementation\" argument. Default \"slim-psr7\" used instead.");
+        }
+    }
+
+    /**
+     * Returns PSR-7 implementation package.
+     *
+     * @return PSR-7 implementation package
+     */
+    public String getPsr7Implementation() {
+        return this.psr7Implementation;
+    }
+
+    /**
+     * Add new package to the `composer.json`.
+     *
+     * @param packageName    Package vendor and name
+     * @param packageVersion Package version e.g. ^1.0.0, default "*"
+     * @param isDev          Add packages to `require-dev`
+     */
+    protected void addComposerPackage(String packageName,
+                                      String packageVersion,
+                                      Boolean isDev) {
+        Map<String, String> composerPackage = new HashMap<String, String>();
+        composerPackage.put("vendorName", packageName);
```

Comment:
I've added `vendorName` and `vendorVersion` codegen properties to add dependencies. The other way was to add specific enum flags `isSlimPsr7`, `isNyholmPsr7`, `isGuzzlePsr7`, `isZendDiactoros` to codegen. Which approach is better?